File: leetcode/easy/1684. Count the Number of Consistent Strings/solution.cc
// https://leetcode.com/problems/count-the-number-of-consistent-strings/submissions/    
#include <iostream>
#include <vector>
#include <string>
#include <unordered_map>

using namespace std;

class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        int count = 0;
        unordered_map<char, int> umap;
        
        for (auto ch: allowed) {
            umap[ch] += 1;
        }

        for (auto w: words) {
            int i = 0;
            for (; i < w.size(); i += 1) {
                if (umap.find(w[i]) == umap.end()) break;
            }
            if (i == w.size()) count += 1;
        }
        cout << count;
        return count;
    }
};

int main() {
    Solution solution;
    string allowed = "ab";
    vector<string> words = {"ad", "bd", "aaab", "baa", "badab"};
    solution.countConsistentStrings(allowed, words);
    return 0;
}

File: leetcode/easy/1684. Count the Number of Consistent Strings/solution.js
// https://leetcode.com/problems/count-the-number-of-consistent-strings/submissions/
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
function countConsistentStrings(allowed, words) {
    let count = 0;
    for (let i = 0; i < words.length; i += 1) {
        for (const w of words[i]) {
            if (!allowed.includes(w)) {
                count += 1;
                break;
            }
        }
    }
    return words.length - count;
};